File: LeetCode/src/20.php
<?php

class Solution
{
  function isValid($s)
  {
    if (strlen($s) % 2 !== 0) {
      return false;
    }
    $stack = [];
    $open = ["(", "{", "["];
    for ($i = 0; $i < strlen($s); ++$i) {
      if (in_array($s[$i], $open)) {
        array_push($stack, $s[$i]);
      } else {
        if ($s[$i] === ")" and end($stack) === "(") {
          array_pop($stack);
        } else if ($s[$i] === "}" and end($stack) === "{") {
          array_pop($stack);
        } else if ($s[$i] === "]" and end($stack) === "[") {
          array_pop($stack);
        } else return false;
      }
    }
    return sizeof($stack) === 0;
  }
}

$test = new Solution();
var_dump($test->isValid("({})[]")); // TEST: bool(true)
var_dump($test->isValid("(){}[]")); // TEST: bool(true)
var_dump($test->isValid("({)}")); // TEST: bool(false)

File: Go/20/20.go
package valid_parentheses

func isValid(s string) bool {
	length := len(s)
	if length%2 != 0 {
		return false
	}

	stack := []rune{}
	for i := 0; i < length; i++ {
		if (s[i] == '(') || s[i] == '{' || s[i] == '[' {
			stack = append(stack, rune(s[i]))
		} else {
			if len(stack) == 0 {
				return false
			}

			if s[i] == ')' && stack[len(stack)-1] == '(' || s[i] == '}' && stack[len(stack)-1] == '{' || s[i] == ']' && stack[len(stack)-1] == '[' {
				stack = stack[:len(stack)-1]
			} else {
				return false
			}
		}
	}

	return len(stack) == 0
}
